Sources/SimpleSalesforceSDK/SSSDK.swift

```diff
@@ -117,34 +117,69 @@ public class SSSDK {
     
     guard let refreshToken = KeychainService.refreshToken else { return }
 
-    WebService.shared.refreshAccessToken(
-      host: host!,
-      clientId: clientId!,
-      clientSecret: clientSecret!,
-      refreshToken: refreshToken
-    )
+    WebService.shared.refreshAccessToken(host: host!,
+                                         clientId: clientId!,
+                                         clientSecret: clientSecret!,
+                                         refreshToken: refreshToken)
   }
 
-  public func fetchData(by query: String, completionHandler: @escaping ((Data?) -> Void)) throws {
+  /// Fetches data using SOQL query
+  /// - Parameters:
+  ///     - query: SOQL query to fetch the data.
+  ///     - completionHandler: Completion handler called when data fetch succeeds `data` is the optional Data from the salesforce.
+  /// - Returns: data for requested SOQL query.
+  /// - Throws: `ConfigurationError.runtimeError` if the singleton is missing the required configuration
+  public func fetchData(by query: String,
+                        completionHandler: @escaping ((Data?) -> Void))
+  throws {
     try! confirmConfiguration()
     
     guard let accessToken = KeychainService.accessToken else {
       throw ConfigurationError.runtimeError("Access Token missing")
     }
-    
     guard let refreshToken = KeychainService.refreshToken else {
       throw ConfigurationError.runtimeError("Refresh Token missing")
     }
     
-    WebService.shared.fetchData(
-      host: host!,
-      clientId: clientId!,
-      clientSecret: clientSecret!,
-      refreshToken: refreshToken,
-      accessToken: accessToken,
-      query: query
-    ) { data in
-      completionHandler(data)
+    WebService.shared.fetchData(host: host!,
+                                clientId: clientId!,
+                                clientSecret: clientSecret!,
+                                refreshToken: refreshToken,
+                                accessToken: accessToken,
+                                query: query,
+                                completionHandler: completionHandler)
+  }
+
+  /// Updates salesforce record using sObject
+  /// - Parameters:
+  ///     - objectName: Object name to update record.
+  ///     - objectId: Record id to update record.
+  ///     - fieldUpdates: Update record data.
+  ///     - completionHandler: Completion handler called when data fetch succeeds `data` is the optional Data from the salesforce.
```

Comment:
```suggestion
  ///     - completionHandler: Completion handler called when data update succeeds in salesforce.
```

Also, do we get data back on update?